tests/conftest.py

```diff
@@ -12,3 +21,240 @@ def api_test_config() -> APITestSessionConfig:
+            "requestLimit_general": "100",
+            "burstLimit_referral": "1200pm",
+            "burstLimit_general": "1200pm"
+        }
+    )
+
+    yield product
+
+    # Teardown
+    print(f"Cleanup product: {product.name}")
+    await product.destroy_product()
+
+
+@pytest.fixture
+async def patient_care_app(
+    make_app,
+    patient_care_product
+):
+    # Setup
+    app = await make_app(
+        patient_care_product,
+        {
+            "jwks-resource-url": __JWKS_RESOURCE_URL
+        },
+    )
+
+    yield app
+
+    # Teardown
+    print(f"Cleanup app: {app.name}")
+    await app.destroy_app()
+
+
+@pytest.fixture
+async def nhs_login_subject_token(patient_care_app: ApigeeApiDeveloperApps):
+    id_token_claims = {
+        "aud": "tf_-APIM-1",
+        "id_status": "verified",
+        "token_use": "id",
+        "auth_time": 1616600683,
+        "iss": "https://internal-dev.api.service.nhs.uk",  # Points to internal dev -> testing JWKS
+        "sub": "https://internal-dev.api.service.nhs.uk",
+        "exp": int(time()) + 300,
+        "iat": int(time()) - 10,
+        "vtm": "https://auth.sandpit.signin.nhs.uk/trustmark/auth.sandpit.signin.nhs.uk",
+        "jti": str(uuid4()),
+        "identity_proofing_level": "P9",
+        "birthdate": "1939-09-26",
+        "nhs_number": "9912003888",
+        "nonce": "randomnonce",
+        "surname": "CARTHY",
+        "vot": "P9.Cp.Cd",
+        "family_name": "CARTHY",
+    }
+
+    id_token_headers = {
+        "kid": "nhs-login",
+        "typ": "JWT",
+        "alg": "RS512"
+    }
+
+    # private key we retrieved from earlier
+    nhs_login_id_token_private_key_path = get_env("ID_TOKEN_NHS_LOGIN_PRIVATE_KEY_ABSOLUTE_PATH")
+
+    with open(nhs_login_id_token_private_key_path, "r") as f:
+        contents = f.read()
+
+    id_token_jwt = patient_care_app.oauth.create_id_token_jwt(
+        algorithm="RS512",
+        claims=id_token_claims,
+        headers=id_token_headers,
+        signing_key=contents,
+    )
+
+    return id_token_jwt
+
+@pytest.fixture
+async def patient_access_token(patient_care_app: ApigeeApiDeveloperApps, nhs_login_subject_token):
+
+    oauth_proxy = get_env("OAUTH_PROXY")
+    oauth_base_uri = get_env("OAUTH_BASE_URI")
+    token_url = f"{oauth_base_uri}/{oauth_proxy}/token"
+
+    jwt = patient_care_app.oauth.create_jwt(
+        **{
+            "kid": "test-1",
+            "claims": {
+                "sub": patient_care_app.client_id,
+                "iss": patient_care_app.client_id,
+                "jti": str(uuid4()),
+                "aud": token_url,
+                "exp": int(time()) + 60,
+            },
+        }
+    )
+
+    token = await get_token(patient_care_app, nhs_login_subject_token, jwt, grant_type="client_credentials")
+    return token["access_token"]
+
+
+async def get_token(
+    app: ApigeeApiDeveloperApps, nhs_login_subject_token, jwt, grant_type: str = "token_exchange"
```

Comment:
Is `grant_type` used here? It looks like the grant type is hard coded to `token-exchange` so perhaps this parameter isn't needed?